src/mo_frontend/typing.ml

```diff
@@ -114,6 +114,27 @@ let kind_of_field_pattern pf = match pf.it with
   | { id; pat = { it = VarP pat_id; _ } } when id = pat_id -> Scope.FieldReference
   | _ -> Scope.Declaration
 
+(* Suggestions *)
+
+let suggest id ids =
+  let rec log2 = function
+    | 1 -> 0
+    | n -> 1 + log2 ((n + 1) / 2) in
+  let limit = log2 (String.length id) in
+  let distance = Lib.String.levenshtein_distance id in
+  let weighted_ids = List.filter_map (fun id0 ->
+    let d = distance id0 in
+    if Lib.String.starts_with id id0 || d <= limit then
+      Some (d, id0)
+    else None) ids in
+  let suggestions = List.sort compare weighted_ids |> List.map snd in
+  if suggestions = [] then ""
+  else
+    let ids, id = Lib.List.split_last suggestions in
+    "\nDid you mean " ^
+      (if ids <> [] then (String.concat ", " ids) ^ " or " else "") ^
```

Comment:
Just a detail: Maybe a unit test for multiple suggestion matches would be interesting.

Reply:
I think there are some (Prim.stableM ) produces several..